File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums.sort()
        n = len(nums)
        s = sum(nums)
        sequence_sum = 0
        i = n
        res = []
        while (sequence_sum <= s):
            i -= 1
            s -= nums[i]
            sequence_sum += nums[i]
            res.append(nums[i])
        return res

File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.cpp
class Solution {
public:
    vector<int> minSubsequence(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int s = 0;  // sum of non selected integers
        for (int i=0; i<n; i++) {
            s += nums[i];
        }
        int sequence_sum = 0;
        int i = n;
        vector<int> res;
        while (sequence_sum <= s) {
            i--;
            s -= nums[i];
            sequence_sum += nums[i];
            res.push_back(nums[i]);
        }
        return res;
    }
};